src/devices/Mcp3xxx/Mcp3xxx.csproj

```diff
@@ -2,15 +2,17 @@
 
   <PropertyGroup>
     <TargetFramework>netcoreapp2.1</TargetFramework>
-    <EnableDefaultItems>false</EnableDefaultItems> <!--Disabling default items so samples source won't get build by the main library-->
+    <!--Disabling default items so samples source won't get build by the main library-->
+    <EnableDefaultItems>false</EnableDefaultItems>
```

Comment:
I think the default items are `<Compile Include="*.cs" />` so you might wanna try removing this as well as `Compile` item

Reply:
If I remove the `<EnableDefaultItems>` then it tries to build the samples recursively as far as I can tell and thus includes the samples.